### ISSUE TITLE
Cannot read property 'arity' of undefined

### ISSUE THREAD

**Issue by flip111** [elm-lang/elm-compiler#1103]
First i had
```
onClick address <| Select row.id
```

Then i changed it to
```
Html.Events.onWithOptions "click" {stopPropagation = True, preventDefault = False} targetValue (\val -> Signal.message address <| Select row.id)
```

And javascript make an error because `A4` is called with `$Html$Events.onWithOptions` being undefined (passed as first argument.

**Comment by evancz** [comment]
Is this with the 0.16 alpha, or with 0.15.1?

**Comment by flip111** [comment]
0.15.1

**Comment by evancz** [comment]
Okay, this is probably not an issue with the compiler, but instead with evancz/elm-html or more likely with evancz/virtual-dom. Can you share a [minimal example](http://sscce.org/) of this? Based on what you are saying, I cannot easily get this set up or see if there are other factors in the program that are needed to make this happen.

**Comment by flip111** [comment]
```elm
import StartApp exposing (start)
import Effects
import Html exposing (..)
import Html.Events exposing (targetValue)

type Action
  = Select Int

update action model = model

view address model =
  div [Html.Events.onWithOptions "click" {stopPropagation = True, preventDefault = False} targetValue (\val -> Signal.message address <| Select 1)]
      []

app = start
        { init = (1, Effects.none)
        , update = (\x y -> (update x y, Effects.none))
        , view = view
        , inputs = []
        }

main = app.html
```

**Comment by evancz** [comment]
I am unable to get this to throw an error, whether it is locally with the 0.16 alpha or on the elm-lang.org/try page. So I am going to close this here as I am not able to reproduce it *and* the root issue is probably in virtual-dom or elm-html.

If you are still able to reproduce it, I need you to tell me the following things in a new issue:

  * What OS are you on?
  * What browser are you using?
  * Do you see this in other browsers?
  * When you get an error, can you share as much information as possible about it? What does it say on the console? Is it on the screen? I don't really even know if I'm on the right track because it is just working fine for me. Maybe there is some specific sequence of events I need to do.

I think I'll have an easier time reproducing and solving the issue that way.

**Comment by flip111** [comment]
Maybe you fixed something in 0.16? I didn't use this version.

* Ubuntu 15.10
* Chromium Version 45.0.2454.101 Ubuntu 15.10 (64-bit)
* Yes Firefox 41.0.2, but with different error message `fun is undefined`
* elm-stuff/exact-dependencies.json
```
{
    "evancz/elm-effects": "2.0.0",
    "circuithub/elm-list-extra": "3.7.0",
    "evancz/virtual-dom": "2.1.0",
    "evancz/start-app": "2.0.1",
    "evancz/elm-html": "4.0.1",
    "elm-lang/core": "2.1.0"
}
```
* I see these messages when viewing the page and also in the console (both chrome and firefox).

I just noticed when i put it in a new project it works for me too. So i check what is the difference with the other project and there i used a html file. So when i start elm-reactor and navigate to this file things go wrong, but to Main.elm directly it's ok.

```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
  </head>
  <body>
    <script src="elm.js"></script>
    <script src="/_reactor/debug.js"></script>
    <script type="text/javascript">
var app = Elm.fullscreenDebug('Main', Elm.Main);
    </script>
  </body>
</html>
```

**Comment by evancz** [comment]
Ah, okay, I think I know the issue then! It is described at https://github.com/elm-lang/elm-reactor/issues/104

`elm-reactor` currently bundles a bunch of Elm code, which attaches thing to the global `Elm` object. Due to #826, these can clobber each other in bad ways. So reactor was built with an unpatched version of elm-html or virtual-dom and while it did not run into the buggy code, it clobbers users new code and make sit bad.

In other words, we have issues open tracking both halves of this and they are well understood, so things should be all set!

Thanks for providing further information, that helped a ton!

**Comment by bharrisau** [comment]
Just ran into this to. `onWithOptions` was introduced in `elm-html` 4.0.0, and `elm-reactor` was using `"evancz/elm-html": "3.0.0 <= v < 4.0.0"`. Reactor 0.16 uses an updated elm-html.

**Comment by flip111** [comment]
I just wanna say on this issue that there are quite a lot of new people who saw the elm debugger video's and then they start coding and the debugger breaks down because of this issue and then they don't know what to do ...
